### DISCUSSION TITLE
[Question][Hw5] About Instruction 3

### DISCUSSION THREAD

**Discussion by kws1207** [prosyslab-classroom/cs348-information-security#600] [Q&A]
I can't understand the sentence (maybe because of my poor english): 

- "Semantic function filter neg cond memory will return an abstract memory that soundly subsumes all possible concrete memories satisfiying condition cond (if neg is true, then the negation of cond)."

, especially the part inside the parenthesis.

Can someone please explain more clearly, maybe with some examples?

**Comment by RiceBiscuits** [comment]
It seems that the instruction 2 has more detail explanation.

> You will define the filter function for memories (Semantics.Make.filter) using the ones for values. The first argument of the function represents whether the condition (i.e., the second argument) is true or not. For example, filter (x < 10) true mem will return an abstract memory that satisfies condition x < 10. On the other hand, filter (x < 10) false mem will return an abstract memory that does not satisfy condition x < 10.

**Comment by kws1207** [comment]
Thanks
